compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -579,6 +587,7 @@ BLangConstantValue constructBLangConstantValueWithExactType(BLangExpression expr
 
         this.anonTypeNameSuffixes = anonTypeNameSuffixes;
         updateConstantType(constantSymbol, expression, env);
+        reset();
```

Comment:
Can u briefly explain why we need to reset here insted of calling at the end of `resolve` method?

Reply:
After the symbolEnter revamp we replace the usage of constantValueResolver for constants with constantTypeChecker. So, constant annotations are only using constantValueResolver. The method `constructBLangConstantValueWithExactType` is invoked in the `symbolResolver` when resolving the const-annotation. Hence I add a reset() end of that method. Actually `resolve` method is not used at all. We have to replace the usage of this class with `constantTypeChecker` as well.

Reply:
Got it shall we call `reset()` method in https://github.com/ballerina-platform/ballerina-lang/blob/4e7e1f4d3c81a0acff294404c39b065efbc54da4/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java#L137 as well and remove single clear statements

Reply:
Also let's rename the method to something like `clearTempData` or similar